### PR TITLE
Moves model

### PR DESCRIPTION
**Pull request by cthao04**

### CODE REVIEW

File: spec/controllers/chess_pieces_controller_spec.rb [hunk 1]
```diff
@@ -3,15 +3,26 @@
...
+
       king = FactoryGirl.create(:king)
-      expect(king.game_id).to eq(1)
       expect(king.user_id).to eq(1)
       expect(king.x).to eq(3)
       expect(king.y).to eq(0)
       expect(king.captured).to eq(false)
       expect(king.type).to eq('King')
     end
+
```
Comment (Muzzh) [inline spec/controllers/chess_pieces_controller_spec.rb]: this commented out section was removed earlier on master

File: app/models/chess_piece.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class ChessPiece < ApplicationRecord
-  # Will have to add belongs_to :game, :user
+  belongs_to :game
```
Comment (Muzzh) [inline app/models/chess_piece.rb]: this should have
`belongs_to :user`
you will probably have to adjust the factories and test for King
```
  factory :king do
    game
    type      'King'
    x         3
    y         0
    captured  false
  end
```
then the test will need adjustment cause we now require a user to exist
```    it "should create dummy King" do
      user = FactoryGirl.create(:user)
      king = FactoryGirl.create(:king, user_id: user.id)

      expect(king.user_id).to eq(user.id)
      expect(king.x).to eq(3)
      expect(king.y).to eq(0)
      expect(king.captured).to eq(false)
      expect(king.type).to eq('King')
```

File: app/models/chess_piece.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
 
+  self.inheritance_column = :type
```
Comment (Muzzh) [inline app/models/chess_piece.rb]: this was actually removed a few commits ago

File: spec/controllers/chess_pieces_controller_spec.rb [hunk 4]
```diff
@@ -3,15 +3,26 @@
 RSpec.describe ChessPiecesController, type: :controller do
   
   describe 'chess_pieces#create' do
+    # let(:game) { FactoryGirl.create(:game) }
```
Comment (Muzzh) [inline spec/controllers/chess_pieces_controller_spec.rb]: these 2 comments can go

Comment (cthao04) [reply spec/controllers/chess_pieces_controller_spec.rb]: Okay, I deleted it.

File: spec/controllers/chess_pieces_controller_spec.rb (hunk 1 above)
Comment (cthao04) [reply spec/controllers/chess_pieces_controller_spec.rb]: Deleted this one as well.

File: app/models/chess_piece.rb (hunk 2 above)
Comment (cthao04) [reply app/models/chess_piece.rb]: Create I add belongs_to :user and adjusted the test as:
  factory :king do
    user
    game
    type      'King'
    user_id   1
    x         3
    y         0
    captured  false
  end

and adjusted the test as you mentioned above.

File: app/models/chess_piece.rb (hunk 3 above)
Comment (cthao04) [reply app/models/chess_piece.rb]: Deleted.

File: app/models/chess_piece.rb (hunk 2 above)
Comment (cthao04) [reply app/models/chess_piece.rb]: I mean the page_controller test.

**Pull request merged**